src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -392,11 +407,48 @@ namespace winrt::Microsoft::Terminal::Control::implementation
                 return true;
             }
 
-            if (ch == Enter)
+            // If we have a connection info with which to try and re-create the
+            // connection, then let's try doing that on Enter
+            if (_ConnectionInfo != nullptr)
             {
-                _connection.Close();
-                _connection.Start();
-                return true;
+                if (ch == Enter)
+                {
+                    // Manually update the startingDirectory. If we had a CWD set
+                    // by the client, then let's try and use that as the CWD for
+                    // a restart, so it's more "seamless"
+                    _ConnectionInfo.Settings().Insert(L"startingDirectory", Windows::Foundation::PropertyValue::CreateString(WorkingDirectory()));
+                    // pass in the magic "inheritCursor" setting to the
+                    // connection's settings. This'll cause conpty to restart
+                    // the connection at the current place in the buffer.
+                    _ConnectionInfo.Settings().Insert(L"inheritCursor", Windows::Foundation::PropertyValue::CreateBoolean(true));
+                    auto c = TerminalConnection::ConnectionInformation::CreateConnection(_ConnectionInfo);
+
+                    // Get our current size in rows/cols, and hook them up to
+                    // this connection too.
+                    {
+                        auto cx = gsl::narrow_cast<til::CoordType>(_panelWidth * _compositionScale);
+                        auto cy = gsl::narrow_cast<til::CoordType>(_panelHeight * _compositionScale);
+                        cx = std::max(cx, _actualFont.GetSize().width);
+                        cy = std::max(cy, _actualFont.GetSize().height);
+                        const auto viewInPixels = Viewport::FromDimensions({ 0, 0 }, { cx, cy });
+                        const auto vp = _renderEngine->GetViewportInCharacters(viewInPixels);

```

Comment:
yikes, we don't know our size any other way???

Reply:
Yes, this should use `_terminal->GetViewport()`.